Feature: I made the home page and about page of a conference website

TracyMuso opened the pull request:
Note: I used other branches to make some other features but I will mention all the changes here.

- I first created the mobile layout of both the home and the about page.
- I used javascript, html and css to implement the slide in toggle menu.
- I then went ahead and styled the desktop version using media queries.
- I dynamically created the featured speakers section using javascript.

TracyMuso committed:
Update menu.js

TracyMuso committed:
Add featured speakers

TracyMuso committed:
fix linter errors

TracyMuso committed:
fix errors

TracyMuso committed:
Add title

TracyMuso committed:
Rename home.html to index.html

TracyMuso committed:
Adjust menu size

TracyMuso committed:
Update READme

TracyMuso committed:
Add screenshots via upload

TracyMuso committed:
Add screenshots

TracyMuso committed:
Update READme

TracyMuso committed:
Add background image

TracyMuso committed:
fix linter errors

TracyMuso committed:
Update about.css

TracyMuso committed:
Update about.css

vechicin commented on home.css:
```diff
@@ -1,6 +1,10 @@
+  scroll-behavior: smooth;
+}
+
 body {
```
There's a horizontal scroll in both versions that makes for bad UI/UX. I'm using Google Chrome for reference.

<img width="228" alt="Screen Shot 2022-07-27 at 9 02 56" src="https://user-images.githubusercontent.com/82426682/181266906-a1eb6725-71b9-4a71-9526-64ef2f10f92d.png">

Please fix it.

vechicin commented on about.css:
```diff
@@ -1,6 +1,10 @@
+  scroll-behavior: smooth;
+}
+
 body {
```
There's a horizontal scroll in both versions that makes for bad UI/UX. I'm using Google Chrome for reference.

<img width="230" alt="Screen Shot 2022-07-27 at 9 04 00" src="https://user-images.githubusercontent.com/82426682/181267164-fe05574c-271e-4e96-87f4-56494b639442.png">

Please fix it.

vechicin commented on home.css:
```diff
@@ -166,6 +186,22 @@ button {
   z-index: 50;
   background-color: #cc8c85;
```
According to the project requirements, the mobile menu's background color should be different than the one you currently have.

<img width="571" alt="Screen Shot 2022-07-27 at 9 06 18" src="https://user-images.githubusercontent.com/82426682/181267794-6e20c39a-e0aa-4981-b6ae-bfd5238c04a4.png">

Please fix the background color for both pages.

vechicin commented on about.css:
```diff
@@ -409,11 +437,12 @@ a {
+    padding: 2rem 0.5rem;
   }
 
   .foot-text {
```
There is a margin at the bottom of your `About` page footer that makes for bad UI/UX.

<img width="1440" alt="Screen Shot 2022-07-27 at 9 11 21" src="https://user-images.githubusercontent.com/82426682/181268942-831d73f8-7958-4745-a2f4-93ae8d987438.png">

Please fix it.

vechicin commented on index.html:
```diff
@@ -54,7 +54,7 @@ <h2>2022.08.27(THU)~30(FRI)</h2>
         </div>
     </section>
-    <section class="program-dets">
+    <section id="program-dets">
```
This section is missing the divider right below the header and the text at the bottom.

<img width="1044" alt="Screen Shot 2022-07-27 at 9 14 48" src="https://user-images.githubusercontent.com/82426682/181269738-90ec8470-4044-4adb-9929-b5eb6570425c.png">

vechicin commented on index.html:
```diff
@@ -123,9 +123,10 @@ <h5 class="topic1">Exhibition</h5>
+        <button class="button">Join BC World Conference 2022</button>
         <a class="p-link" href="#">See whole program</a>
     </section>
+    <section id="fsp"></section>
```
This section is missing the divider below the section header as well.

<img width="1042" alt="Screen Shot 2022-07-27 at 9 17 14" src="https://user-images.githubusercontent.com/82426682/181270336-526cc5a9-2a61-4498-ba5a-1628d159580a.png">

vechicin commented on about.html:
```diff
@@ -64,7 +64,7 @@ <h1 class="main-title">Blockchain World Conference 2022</h1>
+                <strong>bcconference2022@bcaustralia.org</strong>
             </small>
         </div>
         </section>
```
The section right below is missing the divider below the header.

<img width="912" alt="Screen Shot 2022-07-27 at 9 20 25" src="https://user-images.githubusercontent.com/82426682/181271116-7c1f2377-6747-48e0-abd3-d6229e98e6eb.png">

vechicin commented on about.html:
```diff
@@ -83,7 +83,7 @@ <h4>Blockckhain world Conference logo 2022</h4>
         <section class="past">
```
The section right below is missing the divider below the header.

<img width="991" alt="Screen Shot 2022-07-27 at 9 21 51" src="https://user-images.githubusercontent.com/82426682/181271471-033dd1ce-fa0e-4274-8ee2-f66ebdc4080b.png">

vechicin commented on about.css:
```diff
@@ -1,6 +1,10 @@
 @import url('https://fonts.googleapis.com/css2?family=Poppins&display=swap');
```
According to the project requirements, you need to use the `Cocogoose` and `Lato` fonts, or at least fonts that are similar to them.

<img width="465" alt="Screen Shot 2022-07-27 at 9 25 32" src="https://user-images.githubusercontent.com/82426682/181272422-e98b25f3-e564-4ba6-b309-471d76e1ad08.png">

Please fix it.

vechicin commented on index.html:
```diff
@@ -54,7 +54,7 @@ <h2>2022.08.27(THU)~30(FRI)</h2>
+    <section id="program-dets">
         <h3>Main Program</h3>
         <div class="grid">
         <div class="p-grid">
```
Please make sure all of your articles for both pages use their corresponding semantic tag.

<img width="779" alt="Screen Shot 2022-07-27 at 9 27 40" src="https://user-images.githubusercontent.com/82426682/181272960-d335041b-4437-451a-b71c-05524de002ae.png">

vechicin commented on about.html:
```diff
@@ -24,16 +24,16 @@
         </ul>
         </div>
-        <a href="home.html"><img class="nav-logo" src="img/blockchain-logo.png"></a>
+        <a href="index.html"><img class="nav-logo" src="img/blockchain-logo.png"></a>
```
None of your images have alternative text.

<img width="836" alt="Screen Shot 2022-07-27 at 9 29 33" src="https://user-images.githubusercontent.com/82426682/181273490-fcde473f-3b45-48d1-a733-0f17de2774d4.png">

Please add it to all your images.

vechicin commented on feature.js:
```diff
@@ -0,0 +1,108 @@
+       <div class="hold activ">
+            <div class="speaker">
+               <img class="spk-img" src="${spkArr[2].img}">
+               <div class="spk-text">
+                  <h3 class="name">${spkArr[2].name}</h3>
+                  <i class="position">${spkArr[2].position}</i>
+                  <p class="role">${spkArr[2].role}</p>
+               </div>
+            </div>
+            <div class="speaker">
+               <img class="spk-img" src="${spkArr[3].img}">
+               <div class="spk-text">
+                  <h3 class="name">${spkArr[3].name}</h3>
+                  <i class="position">${spkArr[3].position}</i>
+                  <p class="role">${spkArr[3].role}</p>
+               </div>
+            </div>
+       </div>
+
+       <div class="hold activ">
+            <div class="speaker">
+               <img class="spk-img" src="${spkArr[4].img}">
+               <div class="spk-text">
+                  <h3 class="name">${spkArr[4].name}</h3>
+                  <i class="position">${spkArr[4].position}</i>
+                  <p class="role">${spkArr[4].role}</p>
+               </div>
+            </div>
+            <div class="speaker">
+               <img class="spk-img" src="${spkArr[5].img}">
+               <div class="spk-text">
+                  <h3 class="name">${spkArr[5].name}</h3>
+                  <i class="position">${spkArr[5].position}</i>
+                  <p class="role">${spkArr[5].role}</p>
+               </div>
+            </div>
+       </div>
+   </div>
+    `;
+});
```
You should have a function that iterates over your `spkArr` so that it dynamically renders the speaker's information.

<img width="862" alt="Screen Shot 2022-07-27 at 9 33 47" src="https://user-images.githubusercontent.com/82426682/181274556-bfc6eff6-0312-44fd-a436-a45866da791a.png">

Please refactor your code so it iterates over the array mentioned above.

vechicin reviewed (changes requested):
# Hi @elfin-git ✋, 

Good job so far! 🔥 
There are some issues that you still need to work on to go to the next project but you are almost there! 🚨 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights ⚡ 

- [x] Good job deploying the initial version of your Capstone Project!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 📂 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

TracyMuso committed:
fix margin and add alt texts to index.html

TracyMuso commented:
I have looked at all the changes the last reviewer requested and I have implemented some.
**Changes**

- I have added alt texts to all the images
- I removed the horizontal scroll by adding `overflow-x: none;` to the body;
- Both pages now have the same background for the nav bar, which is the background image I put.
- I have used more semantic tags where necessary.

**Things I wont be changing**
-I followed most of the design requirements but I believe that some changes requested should be optional. There are a 1000 ways to design a website, I chose a certain way, I will make those changes later on if I wish to.  Below is the list of the changes I wont be making.

- The orange divider right below the some headings, I don't think that requirement is enough to stop the approval of this pull request.
- The font, I tried cocogoose at first, but the free version I found was much too thick. So I found an alternative, thinner font, and I preferred it. BTW both cocogoose and lato are paid fonts if you want different weights and styles.
- The making of the featured speakers section: I made the featured speakers section dynamically just as the project required me too. I will try out the method the other reviewer asked in my own free time.

**Thankyou for your time! I hope you understand**

TracyMuso committed:
update about.html

TracyMuso committed:
Update home.css

TracyMuso committed:
update index.html

TracyMuso committed:
fix linter error

Nemwel-Boniface commented on feature.js:
```diff
@@ -0,0 +1,108 @@
+       <div class="hold activ">
+            <div class="speaker">
+               <img class="spk-img" src="${spkArr[2].img}" alt="speaker three">
+               <div class="spk-text">
+                  <h3 class="name">${spkArr[2].name}</h3>
+                  <i class="position">${spkArr[2].position}</i>
+                  <p class="role">${spkArr[2].role}</p>
+               </div>
+            </div>
+            <div class="speaker">
+               <img class="spk-img" src="${spkArr[3].img}" alt="speaker four">
+               <div class="spk-text">
+                  <h3 class="name">${spkArr[3].name}</h3>
+                  <i class="position">${spkArr[3].position}</i>
+                  <p class="role">${spkArr[3].role}</p>
+               </div>
+            </div>
+       </div>
+
+       <div class="hold activ">
+            <div class="speaker">
+               <img class="spk-img" src="${spkArr[4].img}" alt="speaker five">
+               <div class="spk-text">
+                  <h3 class="name">${spkArr[4].name}</h3>
+                  <i class="position">${spkArr[4].position}</i>
+                  <p class="role">${spkArr[4].role}</p>
+               </div>
+            </div>
+            <div class="speaker">
+               <img class="spk-img" src="${spkArr[5].img}" alt="speaker six">
+               <div class="spk-text">
+                  <h3 class="name">${spkArr[5].name}</h3>
+                  <i class="position">${spkArr[5].position}</i>
+                  <p class="role">${spkArr[5].role}</p>
+               </div>
+            </div>
+       </div>
+   </div>
+    `;
+});
```
- [ ] You have done an amazing job with trying to generate the speaker's section. However, as the [previous reviewer](https://github.com/elfin-git/Capstone-project-1/pull/4#discussion_r931142502) mentioned above and also according to the project requirements (screenshot below) you are required to generate this section dynamically. Currently, this is not the case. Kindly fix this.

![image](https://user-images.githubusercontent.com/86318284/181300029-07dd53f4-9ea2-4336-a203-317ad8792c16.png)

As you have everything already set up, I had written starter code for another student showing him a very basic example of how to generate the sections dynamically which you can have a look at [here](https://github.com/rebaz36/Portfolio/pull/12#discussion_r925688836). Kindly make this change and your project should be ready for approval.

Nemwel-Boniface reviewed (changes requested):
## Hello @elfin-git           :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Great job addressing some of the changes that were requested by a previous reviewer :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!

![](https://media.giphy.com/media/KyBesf8kNT6s88SEgb/giphy.gif)
## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

Kindly note that the changes that the reviewers request you to do are supposed to help you better your project so that when it reaches the time when you want to display your projects to someone, they are awed by your work.

You should also see yourself in a client's shoes where you tell him that you will not make a required change mainly because you have different ways of implementing things but the client has his own set standards which you do not want to follow. This will leave you jobless with a bad record. This said, I will reiterate some of the changes requested by the previous reviewer.

- [ ] You can use this import to import cocogoose to your project and make the changes that were requested on using the correct fonts as per the project requirements. ```@import url('http://fonts.cdnfonts.com/css/cocogoose');```. **Make sure to make this change because the next reviewer will assert that you have used the correct fonts as per my requested changes.**

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

TracyMuso committed:
Create speakers dynamically

TracyMuso committed:
Update css files

TracyMuso committed:
add cocogoose font

TracyMuso commented:
I added the cocogoose fonts and created the speakers feature dynamically using functions and iterating through an array.

sja-thedude reviewed (approved):
## Project Approved 👍 

Hi @elfin-git, This is @sja-thedude your code reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

![giphy](https://user-images.githubusercontent.com/70269507/159485323-11dfc0c5-bd53-4956-b55a-2fc42f7a3845.gif)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
